### PR TITLE
Introducing link feature (from v1.3.11 to v1.3.12)

### PR DESCRIPTION
**Pull request by richardevcom**
Introducing link feature:

- Automatically convert text containing link URL to clickable link element
- Select text and create link element
- Edit existing link element
- Remove all selected URLs from link elements

preview

@lostdesign you were too fast and I couldn't create pull request fro v1.3.11 tagged to my v1.3.12 tagged version, so I am requesting this on your main branch 😆

### CODE REVIEW

**Review by lostdesign — CHANGES_REQUESTED** [review]
Judging based on the gif, it'd be better if the link opened directly on click don't you think?

**Comment by richardevcom** [comment]
> Judging based on the gif, it'd be better if the link opened directly on click don't you think?

```js
Link({
  openOnClick: false,
})
```
is not extendable, so we need `on` `click` event listener directly in tiptap editor.. which honestly I didn't find in their solution. My guess it will have to be done within Vue.js scope.

**Comment by richardevcom** [comment]
updated https://github.com/lostdesign/linked/pull/108/commits/7dc7dc8da59bcb0d81852fb56ede458dbcfcc4f5

**Review by lostdesign — CHANGES_REQUESTED** [review]
See the other commands as well as:
Opening directly on click seems a bit weird actually - I think it is better to have cmd/ctrl+click to open them. 

Other than that looks good, I'd contribute to your branch to work on the design as that would need to be adjusted before release. But we've discussed that - I might get to that end of November however.

File: src/assets/css/tailwind.css [hunk 1]
```diff
@@ -29,6 +29,10 @@ mark {
   @apply bg-yellow-200;
 }
 
+.hidden {
```
[the remark this answers is not in the document]
Comment (richardevcom) [reply src/assets/css/tailwind.css:32]: lol. for a moment I think I just didn't switch to `tailwind.config.js` for dev env.

File: src/assets/icons/open-link.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 22 22" height="15px"><path d="M21,11a1,1,0,0,0-1,1v7a1,1,0,0,1-1,1H3a1,1,0,0,1-1-1V3A1,1,0,0,1,3,2h7a1,1,0,0,0,0-2H3A3,3,0,0,0,0,3V19a3,3,0,0,0,3,3H19a3,3,0,0,0,3-3V12A1,1,0,0,0,21,11Z" fill="currentColor"/><path d="M22,1A1.37,1.37,0,0,0,22,.8a.19.19,0,0,1,0-.08.42.42,0,0,0,0-.1l0-.09L21.83.44l-.1-.12,0,0,0,0-.13-.1L21.47.12l-.09,0-.1,0L21.2,0,21,0H15a1,1,0,0,0,0,2h3.59l-8.2,8.19a1,1,0,0,0,1.42,1.42L20,3.41V7a1,1,0,0,0,2,0V1Z" fill="currentColor"/></svg>
```
[the remark this answers is not in the document]
Comment (richardevcom) [reply src/assets/icons/open-link.svg:1]: https://www.iconfinder.com free icon (it has https://creativecommons.org/licenses/by-nc/3.0/ Non Commercial license so.. I guess I will draw my own quickly).

File: src/assets/icons/unlink.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...www.w3.org/2000/svg" viewBox="0 0 22 21.95" height="15"><path d="M18.84,3.24a2.44,2.44,0,0,0-3.46,0L11.93,6.7a1.24,1.24,0,0,1-1.73,0A1.22,1.22,0,0,1,10.2,5l3.45-3.46a4.89,4.89,0,0,1,6.92,6.91l-3.46,3.46a1.22,1.22,0,0,1-1.73,0,1.24,1.24,0,0,1,0-1.73L18.84,6.7A2.46,2.46,0,0,0,18.84,3.24ZM6.62,10.17a1.24,1.24,0,0,1,0,1.73L3.16,15.36a2.44,2.44,0,1,0,3.46,3.45l3.45-3.45a1.22,1.22,0,0,1,1.73,1.73L8.35,20.54a4.89,4.89,0,0,1-6.92-6.91l3.46-3.46A1.22,1.22,0,0,1,6.62,10.17Z" transform="translate(0 -0.03)" fill="currentColor"/><path d="M7.33,0A1.22,1.22,0,0,1,8.56,1.25V2.47a1.23,1.23,0,0,1-2.45,0V1.25A1.22,1.22,0,0,1,7.33,0ZM1.58,1.61a1.22,1.22,0,0,1,1.73,0L4.53,2.83a1.22,1.22,0,0,1,0,1.73,1.24,1.24,0,0,1-1.73,0L1.58,3.33A1.21,1.21,0,0,1,1.58,1.61ZM0,7.36A1.22,1.22,0,0,1,1.22,6.14H2.44a1.22,1.22,0,1,1,0,2.44H1.22A1.22,1.22,0,0,1,0,7.36Zm18.33,6.11a1.23,1.23,0,0,1,1.23-1.22h1.22a1.22,1.22,0,0,1,0,2.44H19.56A1.22,1.22,0,0,1,18.33,13.47Zm-.86,2.8a1.24,1.24,0,0,1,1.73,0l1.22,1.23a1.22,1.22,0,0,1-1.73,1.72L17.47,18A1.22,1.22,0,0,1,17.47,16.27Zm-2.8.87a1.22,1.22,0,0,1,1.22,1.22v1.22a1.23,1.23,0,0,1-2.45,0V18.36A1.22,1.22,0,0,1,14.67,17.14Z" transform="translate(0 -0.03)" fill="currentColor"/></svg>
```
[the remark this answers is not in the document]
Comment (richardevcom) [reply src/assets/icons/unlink.svg:1]: https://www.iconfinder.com free icon (it has https://creativecommons.org/licenses/by-nc/3.0/ Non Commercial license so.. I guess I will draw my own quickly).

File: src/background.js [hunk 4]
```diff
@@ -284,6 +285,10 @@ ipcMain.handle('SAVE_FILE', (event, args) => {
   )
 })
 
+ipcMain.on('openurl', (event, arg) => {
```
[the remark this answers is not in the document]
Comment (richardevcom) [reply src/background.js:288]: On quick look I thought those are Vuex store related methods 👀 will do.

File: src/assets/icons/open-link.svg (hunk 2 above)
[the remark this answers is not in the document]
Comment (lostdesign) [reply src/assets/icons/open-link.svg:1]: No need, we can use iconic.app

**Comment by richardevcom** [comment]
> 
Yeah let's see who from us both will do it first 😄 In future `on` `click` and `cmd` + `click` should be set as setting for user to select from. For now I do agree `cmd` + `click` is more user friendly.

About that layout/UI - I'll ask around for suggestions.

Let me fix suggestions above and "p.r." final-final-final branch 😄

**Comment by lostdesign** [comment]
Closing meanwhile as linked progressed so much.

**Pull request closed without merging**